src/runtime_src/core/pcie/driver/linux/xocl/subdev/mb_scheduler.c

```diff
@@ -4613,7 +4618,8 @@ int cu_map_addr(struct platform_device *pdev, u32 cu_idx, void *drm_filp,
 		mutex_unlock(&xdev->dev_lock);
 		return -EINVAL;
 	}
-	if (ip_excl_holder(exec, cu_idx) == 0) {
+	/* WORKAROUND: If rw_shared is true, allow map shared CU */
+	if (!exec->rw_shared && ip_excl_holder(exec, cu_idx) == 0) {
```

Comment:
do we want to check if user is root?

Reply:
We could easily add that once we have to.